packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2008,6 +1901,21 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
           }
         }
       }
+      if (supportsPersistence) {
+        const fiberRoot: FiberRoot = finishedWork.stateNode;
+        const containerInfo = fiberRoot.containerInfo;
+        const pendingChildren = fiberRoot.pendingChildren;
+        replaceContainerChildren(containerInfo, pendingChildren);
+      }
+      return;
+    }
+    case HostPortal: {
+      if (supportsPersistence) {
+        const portal = finishedWork.stateNode;
+        const containerInfo = portal.containerInfo;
+        const pendingChildren = portal.pendingChildren;
+        replaceContainerChildren(containerInfo, pendingChildren);
+      }
```

Comment:
I think HostPortal used to throw at the bottom (when no cases match) in mutating mode because it's not supposed to have effects on it. Was this a useful invariant or not really?

Reply:
Oh I see you removed that check altogether.

Reply:
Yeah I removed it because I don't think the invariant is that useful